update testsys-config to toml v0.8 or later

Pretty soon testsys will move to this repo, and when it does we need to deal with this in toml 0.7 or later. It is deprecated in 0.6 and removed on 0.7.



```
error[E0425]: cannot find function `tables_last` in module `toml::ser`
  --> testsys-config/src/lib.rs:24:39
   |
24 |     #[serde(flatten, serialize_with = "toml::ser::tables_last")]
   |                                       ^^^^^^^^^^^^^^^^^^^^^^^^ not found in `toml::ser`
```

--- a/deny.toml
+++ b/deny.toml
@@ -89,8 +89,6 @@ skip = [
     { name = "socket2", version = "0.4" },
     # multiple deps are using an older version of syn
     { name = "syn", version = "1" },
-    # Testsys is using a feature of TOML 0.5 that has been removed in subsequent versions.
-    { name = "toml", version = "0.5" },
 
     # aws-sdk-rust is using an old version of rustls, hyper-rustls, and tokio-rustls
     { name = "rustls", version = "=0.20" },

--- a/tools/testsys-config/Cargo.toml
+++ b/tools/testsys-config/Cargo.toml
@@ -17,4 +17,4 @@ serde = { version = "1", features = ["derive"] }
 serde_plain = "1"
 serde_yaml = "0.9"
 snafu = "0.7"
-toml = "0.5"
+toml = "0.8"

--- a/tools/testsys-config/src/lib.rs
+++ b/tools/testsys-config/src/lib.rs
@@ -21,7 +21,7 @@ pub struct TestConfig {
     /// High level configuration for TestSys
     pub test: Option<Test>,
 
-    #[serde(flatten, serialize_with = "toml::ser::tables_last")]
+    #[serde(flatten)]
     /// Configuration for testing variants
     pub configs: HashMap<String, GenericConfig>,
 }

